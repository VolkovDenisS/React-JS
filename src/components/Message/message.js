import React from 'react';
import client from '../img/profile.png';
import {Button} from "react-bootstrap";
import Moment from 'react-moment';
import styles from './MessageList.module.css';
import {deleteMessage} from "../../store/messages/actions";

export const Message = ({message}) => {

  return (
      <div className={`${styles.container}`} key={message.id}>
          <img src={client} alt="Avatar"/>
          <p><span>{message.author}</span>: <span>{message.text}</span></p>
          <span className={styles.timeRight}><Moment unix>{Date.now()}</Moment></span>
      </div>
  );
};



