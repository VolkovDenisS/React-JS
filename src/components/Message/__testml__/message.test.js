import { render } from "@testing-library/react";
import { Message} from "../message";

describe("Message tests", () => {
  it("renders author and text", () => {
    const message = render(<Message message={{ text: "text", author: "volkov.denis.w@gmail.com" }} />);

    const msgText = message.getByText("text");
    const authorText = message.getByText("author");
    expect(msgText).toBeInTheDocument();
    expect(authorText).toBeInTheDocument();
  });

  it("matches snapshot with author", () => {
    const message = render(<Message message={{ text: "text", author: "author" }} />);

    expect(message).toMatchSnapshot();
  });
  it("matches snapshot with no author", () => {
    const message = render(<Message message={{ text: "text", author: "" }} />);

    expect(message).toMatchSnapshot();
  });
});
