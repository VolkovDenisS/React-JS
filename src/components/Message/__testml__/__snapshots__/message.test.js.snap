// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message tests matches snapshot with author 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <img
          alt="Avatar"
          src="profile.png"
        />
        <p>
          <span>
            author
          </span>
          : 
          <span>
            text
          </span>
        </p>
        <span
          class="timeRight"
        >
          <time
            datetime="1645545629828000"
          >
            Mon Apr 08 54115 03:17:08 GMT+0300
          </time>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <img
        alt="Avatar"
        src="profile.png"
      />
      <p>
        <span>
          author
        </span>
        : 
        <span>
          text
        </span>
      </p>
      <span
        class="timeRight"
      >
        <time
          datetime="1645545629828000"
        >
          Mon Apr 08 54115 03:17:08 GMT+0300
        </time>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Message tests matches snapshot with no author 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <img
          alt="Avatar"
          src="profile.png"
        />
        <p>
          <span />
          : 
          <span>
            text
          </span>
        </p>
        <span
          class="timeRight"
        >
          <time
            datetime="1645545639244000"
          >
            Mon Apr 08 54115 05:54:04 GMT+0300
          </time>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <img
        alt="Avatar"
        src="profile.png"
      />
      <p>
        <span />
        : 
        <span>
          text
        </span>
      </p>
      <span
        class="timeRight"
      >
        <time
          datetime="1645545639244000"
        >
          Mon Apr 08 54115 05:54:04 GMT+0300
        </time>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
